File: SystemIntegration/Assignments/01a/FileParser/Pokemon.cs
using System.Text.Json.Serialization;
using System.Xml.Serialization;

[XmlRoot (ElementName = "pokemon")]
public class Pokemon
{
    [JsonPropertyName("nationalNo")]
    [XmlElement ("nationalNo")]
    public int? NationalNo { get; set;}

    [JsonPropertyName("name")]
    [XmlElement ("name")]
    public string? Name { get; set; }

    [JsonPropertyName("types")]
    [XmlArray (ElementName = "types")]
    [XmlArrayItem (ElementName = "type")]
    public List<string>? Types { get; set; }
}

File: SystemIntegration/Assignments/01a/NodeFileParser/Pokemon.js
class Pokemon {
    constructor({ name, nationalNo, types}) {
        this.nationalNo = nationalNo;
        this.name = name;
        this.types = types;
    }

    toString() {
        return `Pokemon: (#${this.nationalNo}), ${this.name}, ${this.types.join(', ')} `
    }
}

module.exports = Pokemon;